lib/puppet/provider/yumrepo/inifile.rb

```diff
@@ -82,10 +82,10 @@ def new_physical_file(file)
   # @api public
   # @return [Array<Puppet::Provider>] providers generated from existing yum
   #   repository definitions.
-  def self.instances
+  def self.instances(resources = nil)
```

Comment:
Since `instances` is a public provider method that overrides https://github.com/puppetlabs/puppet/blob/82ad86ea21ecf20a956db6bf46140940caccd0cc/lib/puppet/provider.rb#L382 it's surprising for this method to have a different signature than its parent. Since the `instances` method may be called in two different contexts (during prefetch or when running `puppet resource yumrepo`) I'd recommend creating a private class method containing all of the old `instances` code and then call it with different arguments from `instances` and `prefetch`, something like:

```suggestion
  def self.instances
    instances_with_resources
  end
  
  def self.instances_with_resources(resources = nil)
    # existing instances code
  end
  private_class_method :instances_with_resources
  
  def self.prefetch(resources)
    repos = instances_with_resources(resources)
    ...
  end
```